java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java

```diff
@@ -740,6 +740,39 @@ private static String getBusId(Terminal t) {
                             .addStringSeries("bus_id", svc -> getBusId(svc.getTerminal())))
                             .build();
 
+                case VOLTAGE_LEVEL:
+                    List<VoltageLevel> voltageLevels = network.getVoltageLevelStream().collect(Collectors.toList());
+                    return addProperties(new SeriesPointerArrayBuilder<>(voltageLevels)
+                            .addStringSeries("id", VoltageLevel::getId)
+                            .addDoubleSeries("nominal_v", VoltageLevel::getNominalV)
+                            .addDoubleSeries("high_voltage_limit", VoltageLevel::getHighVoltageLimit)
+                            .addDoubleSeries("low_voltage_limit", VoltageLevel::getLowVoltageLimit))
+                            .build();
+
+                case SUBSTATION:
+                    List<Substation> substations = network.getSubstationStream().collect(Collectors.toList());
+                    return addProperties(new SeriesPointerArrayBuilder<>(substations))
+                            .addStringSeries("id", Identifiable::getId)
+                            .addStringSeries("tos", Substation::getTso)
+                            .addStringSeries("geo_tags", substation -> String.join(",", substation.getGeographicalTags()))
+                            .addStringSeries("country", substation -> {
+                                final Optional<Country> optCountry = substation.getCountry();
+                                if (optCountry.isPresent()) {
```

Comment:
I guess you can use `map` here instead of if/else.
Also, I think it will be more consistent to return an empty string if the country is not defined: that's what is done so far for buses for example (when an equipment is not connected to any bus).